File: week6/Chapter_7_Organization/javascript/src/domain/geometry/reporting/ShapeReporter.js
export class ShapeReporter {
  measure(shape) {
    return { area: shape.area(), perimeter: shape.perimeter() };
  }

  summary(shapes) {
    let totalArea = 0;
    let totalPerimeter = 0;

    const lines = [];
    let i = 1;

    for (const shape of shapes) {
      const m = this.measure(shape);
      totalArea += m.area;
      totalPerimeter += m.perimeter;
      lines.push(`${i}. ${shape.constructor.name}: area=${m.area.toFixed(2)}, perimeter=${m.perimeter.toFixed(2)}`);
      i += 1;
    }

    lines.push("-");
    lines.push(`TOTAL area=${totalArea.toFixed(2)}, TOTAL perimeter=${totalPerimeter.toFixed(2)}`);
    return lines.join("\n");
  }
}

File: week6/Chapter_7_Organization/csharp/src/OopOrganization.Chapter7/Domain/Geometry/Reporting/ShapeReporter.cs
using OopOrganization.Domain.Geometry.Contracts;

namespace OopOrganization.Domain.Geometry.Reporting;

public sealed class ShapeReporter
{
    public string Summary(IEnumerable<IShape> shapes)
    {
        var lines = new List<string>();
        double totalArea = 0;
        double totalPerimeter = 0;

        int i = 1;
        foreach (var shape in shapes)
        {
            var area = shape.Area();
            var perimeter = shape.Perimeter();
            totalArea += area;
            totalPerimeter += perimeter;

            lines.Add($"{i}. {shape.GetType().Name}: area={area:F2}, perimeter={perimeter:F2}");
            i++;
        }

        lines.Add("-");
        lines.Add($"TOTAL area={totalArea:F2}, TOTAL perimeter={totalPerimeter:F2}");
        return string.Join(Environment.NewLine, lines);
    }
}
